server/genrest/echo.go

```diff
@@ -19,17 +19,28 @@ package genrest
 
 import (
 	"context"
+	"fmt"
 	"net/http"
 
 	"github.com/golang/protobuf/jsonpb"
 	genprotopb "github.com/googleapis/gapic-showcase/server/genproto"
+	gmux "github.com/gorilla/mux"
 )
 
 // HandleEcho translates REST requests/responses on the wire to internal proto messages for Echo
 //    Generated for HTTP binding pattern: /v1beta1/echo:echo
 //         This matches URIs of the form: /v1beta1/echo:echo
 func (backend *RESTBackend) HandleEcho(w http.ResponseWriter, r *http.Request) {
+	urlPathParams := gmux.Vars(r)
+	numUrlPathParams := len(urlPathParams)
+
 	backend.StdLog.Printf("Received request matching '/v1beta1/echo:echo': %q", r.URL)
+	backend.StdLog.Printf("  urlPathParams (expect 0, have %d): %q", numUrlPathParams, urlPathParams)
+
+	if numUrlPathParams != 0 {
+		w.Write([]byte(fmt.Sprintf("unexpected number of URL variables: expected 0, have %d: %#v", numUrlPathParams, urlPathParams)))
+		return
+	}
```

Comment:
It looks like this block is repeated multiple times. How onerous would it be to lift it out into something named e.g. `verifyZeroPathParams`?

Reply:
This is a generated file, so I don't feel too bad about having repeated code. WDYT?

I suppose we could have the generated code call a common (non-generated) function to check for the expected number of params, but then the function call itself would be repeated. I'd like to wait until the generated functionality is more stable before doing that, because there's likely more similar refactoring we could do then.

I add a TODO for now so we don't forget about it.

Thoughts?

Reply:
Ah, if it's generated code then I agree, repeated chunks aren't too problematic.